sklearn/neighbors/_base.py

```diff
@@ -1021,6 +1021,16 @@ class from an array representing our data set and ask who's
                 neigh_ind_list = sum(chunked_results, [])
                 results = _to_object_array(neigh_ind_list)
 
+            if sort_results:
+                if not return_distance:
+                    raise ValueError("return_distance must be True "
+                                     "if sort_results is True.")
+                for ii in range(len(neigh_dist)):
+                    order = np.argsort(neigh_dist[ii], kind='mergesort')
```

Comment:
just curious, why do we need the sort to be stable here?

Reply:
I used mergesort to be consistent with the sorting done in `_check_precomputed`, but I don't remember why we chose it.

Reply:
https://github.com/scikit-learn/scikit-learn/pull/10482#discussion_r167456424